pkg/subctl/cmd/join.go

```diff
@@ -468,5 +482,5 @@ func operatorImage() string {
 		version = versions.DefaultSubmarinerOperatorVersion
 	}
 
-	return images.GetImagePath(repository, version, deployment.OperatorName, nil)
+	return images.GetImagePath(repository, version, deployment.OperatorName, imageOverrides)
```

Comment:
Oh wait, I just realized that this won't work properly if the order of calls is altered.

If this function is called before populateSubmarinerSpec , then imageOverrides will always be empty. I haven't looked at the code, and it could be the other way around, but it would be a hard to catch bug if the order of call is modified in future patches.


Can you remove the global variable "imageOverrides" 

and create a function: 
```
func getImageOverrides() map[string] string {
if len(imageOverrideArr) > 0 {
		imageOverrides := make(map[string]string)
		for _, s := range imageOverrideArr {
			key := strings.Split(s, "=")[0]
			value := strings.Split(s, "=")[1]
			imageOverrides[key] = value
		}
		return imageOverrides
	}
return nil
}
```

And then you can call it here... and also in the populateSubmarinerSpec

It's of course more inefficient, because we calculate this twice. But then the order of the function calls doesn't matter.

Reply:
Done